src/runtime/thread_pool.cc

```diff
@@ -300,18 +300,13 @@ class ThreadPool {
     return dmlc::ThreadLocalStore<ThreadPool>::Get();
   }
 
-  void updateWorkerConfig(int mode, int nthreads) {
-    std::vector<unsigned int> sorted_order;
-    unsigned int num_workers_used = threading::configThreadGroup(mode, nthreads, &sorted_order);
+  void UpdateWorkerConfig(int mode, int nthreads) {
+    unsigned int num_workers_used = threading::ConfigThreadGroup(mode, nthreads, threads_.get());
```

Comment:
Again, combine threading::ConfigThreadGroup and SetAffinity into 

```c++
// rename num_workers_used_ -> num_active_workers_;
num_active_workers_ = threads_->Configure(exclude_worker0, mode, nthreads);
```

Reply:
Change the constructor of ThreadGroup to be simple form, and call threads_->Configure(exclude_worker0, mode, num_workers_); in the constructor